mit-bitcoin-expo/mit-bitcoin-expo-2021/defending-against-99-attack.md

```diff
@@ -0,0 +1,76 @@
+
+Moderator: we have time for about two more questions, the next question is: Converse to what you were saying about reducing the hash size, isn't increasing the hash size supposed to future proof the system instead?
+Tadge: Well, maybe. But you can't. You could say, let's put something new in Bitcoin where we have 512 bit pubkey hashes. Or we use an elliptic curve that 400 or 500 bits long, so we double the security. Well okay, but we still have almost everyone using the previous parts of the system. If somebody makes a computer that can derive a private key from a public key, which takes 2^128 ellitic curve ops. Maybe you're safe, but everyone else isn't, so the whole system isn't. If you're going to make your Bitcoins twice as secure, it doesn't matter, if everybody else's coins have been stolen it doesn't matter because the system has collapsed. It's no use to say "I still have mine", because everybody just leaves the system. So there may be ways to upgrade, but you kind of need everyone to upgrade, or almost everyone. If there's 5 million of the 21 million coins are stealable, the system is in trouble. 
+
+Moderator: What about quantum computers, is Bitcoin ready for that? 
+Tadge: Right now no, definitely not. If you have a quantum computer, you can grab all of Satoshi's coins, there's so many unhashed public keys that are known that havve enormous amounts of coins so you just pick one and perform a quantum operation. You perform that attack and you take all the coins. Right now it's not, it could be. A system like Bitcoin would be pretty amenable to being quantum safe. There are a bunch of quantum safe signature schemes that we could use, but they're worse in that they're bigger and slower and stuff. But you could do it. The question of upgrading is very difficult. You sort of have to change everybody's keys. Even if it's just a quarter of the coins that are stealable, is this system going to survive that? There's also the question of quantum computers and proof of work, how do those interact? So right now no, it's something people are thinking about but it's hard to upgrade everyone. Maybe if a quantum computer comes out or it's imminent, then it's something that's scary enough that nearly everyone will move, and we make some kind of soft fork which says all of these old coins are frozen forever. But it would be very disruptive. We're not really planning for that right now.
```

Comment:
havve
```suggestion
Tadge: Right now no, definitely not. If you have a quantum computer, you can grab all of Satoshi's coins, there's so many unhashed public keys that are known that have enormous amounts of coins so you just pick one and perform a quantum operation. You perform that attack and you take all the coins. Right now it's not, it could be. A system like Bitcoin would be pretty amenable to being quantum safe. There are a bunch of quantum safe signature schemes that we could use, but they're worse in that they're bigger and slower and stuff. But you could do it. The question of upgrading is very difficult. You sort of have to change everybody's keys. Even if it's just a quarter of the coins that are stealable, is this system going to survive that? There's also the question of quantum computers and proof of work, how do those interact? So right now no, it's something people are thinking about but it's hard to upgrade everyone. Maybe if a quantum computer comes out or it's imminent, then it's something that's scary enough that nearly everyone will move, and we make some kind of soft fork which says all of these old coins are frozen forever. But it would be very disruptive. We're not really planning for that right now.
```